carbon-p2-plugin/src/main/java/org/wso2/maven/p2/utils/FileManagementUtil.java

```diff
@@ -0,0 +1,291 @@
+/*
+*  Copyright (c) 2005-2011, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package org.wso2.maven.p2.utils;
+
+import org.apache.maven.plugin.logging.Log;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Properties;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+/**
+ * Util class which handle file manipulation operations.
+ */
+public class FileManagementUtil {
+    private static final int BUFFER = 2048;
+
+
+    /**
+     * For a given location and the profile, returns the File object represented by config.ini
+     *
+     * @param destination path pointing the [carbon product]/repository/components folder
+     * @param profile     name of the profile
+     * @return the config.ini file for the Profile
+     */
+    public static File getProfileConfigIniFile(String destination, String profile) {
+        return new File(destination + File.separator + profile + File.separator + "configuration" +
+                File.separator + "config.ini");
+    }
+
+    /**
+     * Updated the given config.ini file with a given key value pair.
+     *
+     * @param configIniFile File object representing the config.ini
+     * @param propKey       property key
+     * @param value         property value
+     */
+    public static void changeConfigIniProperty(File configIniFile, String propKey, String value, Log log) {
+        Properties prop = new Properties();
+
+        try (InputStream inputStream = new FileInputStream(configIniFile)) {
+            prop.load(inputStream);
+            prop.setProperty(propKey, value);
+            try (OutputStream outputStream = new FileOutputStream(configIniFile)) {
+                prop.store(outputStream, null);
+            }
+        } catch (IOException e) {
+            log.warn(e);
```

Comment:
Shouldn't we propagate the error back to the caller? If the property update was not successful, we need to inform the caller IMO.